src/main/groovy/de/iteratec/osm/measurement/environment/wptserver/WptResultXml.groovy

```diff
@@ -228,18 +232,23 @@ class WptResultXml {
 
     boolean isValidTestStep(GPathResult testStepNode) {
         if (!testStepNode || testStepNode.isEmpty()) {
+            log.error("Step node is empty: ${testStepNode}")
             return false
         }
-        Integer fullyLoaded = testStepNode.fullyLoaded?.toInteger()
-        WptStatus wptStatus = WptStatus.byResultCode(testStepNode.result?.toInteger())
-        Integer ttfb = testStepNode.TTFB?.toInteger()
-        if (ttfb <= 0) {
-            log.warn("Invalid TTFB '${ttfb}' in test:'${getTestId()}', regarding it as valid anyway because of known WPT problems.")
+        int statusCode = testStepNode.result?.toInteger()
+        if (statusCode == null) {
+            log.error("Can't parse status code for step result from wpt: ${testStepNode}")
+            return false
         }
-        boolean valid = wptStatus.isSuccess() && fullyLoaded >= minValidLoadTime && fullyLoaded <= maxValidLoadTime
-        if (!valid) {
-            log.info("Invalid EventResult in test:'${getTestId()}' (Status code: ${wptStatus}, TTFB: ${ttfb}, fullyLoaded: ${fullyLoaded})")
+        if (!WptStatus.byResultCode(statusCode).isSuccess()) {
+            return false
+        }
+        if (!atLeastExpectedToBeValid.any{ measurand ->
```

Comment:
This is too long for a If-statement and should be extracted into a new method.

Reply:
you are right.